lib/kernel/src/group.erl

```diff
@@ -682,6 +671,25 @@ io_reply(From, ReplyAs, Reply) ->
     From ! {io_reply,ReplyAs,Reply},
     ok.
 
+send_drv_latin1_raw(Drv, Chars, From)->
+    Latin1 = get_unicode_state(Drv) =:= false,
+    Raw = get_output_state(Drv) =:= raw,
```

Comment:
Is my understanding correct that this adds two extra round-trip RPC calls between group and the driver for every print request?

Reply:
Yes, your understanding was correct, but that has been removed, I think we can assume that user_drvers output mode is raw when we are in noshell mode. And we then only have to check encoding is latin1. which is now done in setopts. I have changed so that we dont need the extra roundtrips.

Reply:
> I think we can assume that user_drvers output mode is raw when we are in noshell mode.

No, you cannot. The `shell =:= noshell` in `group` is used to determine if a shell process is attached to the specific group, that is if the group is the `user` process or not. If the system has a shell, there will be one or more other `group` processes that handles the shells. Also, the change of encoding request can happen from any of the groups in the system.

You can however assume that it is only the process with `shell =:= noshell` that we can do the optimization with, so if you want to avoid doing the calls to `user_drv`, then `user_drv` needs to notify the `user` `group` of the encoding change. `user_drv` can also switch from being in `raw` to `cooked` mode if anyother issues `shell:start_interactive/0`, so `user_drv` also need to notify `user` of that change.

Reply:
Aha, then I will revert back to asking the user_drv for now. But make it into a single question instead of two.